cmd/minishift/cmd/profile/profile_set.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package profile
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	cmdUtil "github.com/minishift/minishift/cmd/minishift/cmd/util"
+	profileActions "github.com/minishift/minishift/pkg/minishift/profile"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/spf13/cobra"
+)
+
+const (
+	unableToSetOcContextErrorMessage = "Make sure the profile is in running state or restart if the problem persists."
+)
+
+var profileSetCmd = &cobra.Command{
+	Use:   "set PROFILE_NAME",
+	Short: "Sets the active profile for Minishift.",
+	Long:  "Sets the active profile for Minishift. After you set the profile, all commands will use the specified profile by default.",
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) == 0 {
+			atexit.ExitWithMessage(1, emptyProfileMessage)
+		} else if len(args) > 1 {
+			atexit.ExitWithMessage(1, extraArgumentMessage)
+		}
+
+		profileName := args[0]
+
+		if !cmdUtil.IsValidProfile(profileName) {
+			atexit.ExitWithMessage(1, fmt.Sprintf("Error: '%s' is not a valid profile", profileName))
+		}
+
+		err := profileActions.SetActiveProfile(profileName)
+		if err != nil {
+			atexit.ExitWithMessage(1, err.Error())
+		} else {
+			fmt.Println(fmt.Sprintf("Profile '%s' set as active profile", profileName))
```

Comment:
@LalatenduMohanty This message might create confusion with the possible error message by `SetOcContext`.
For example, 
- `SetActiveProfile` passes and it print message `Profile ... set as active profile`
- `SetOcContext` fails and if message is `Error during setting '....' as active profile: ..........`
```go
		return errors.New(fmt.Sprintf("Error during setting '%s' as active profile: %s", profileName, err.Error()))
	}
	err = ocRunner.AddCliContext(constants.MachineName, ip, defaultUser, defaultProject)
	if err != nil {
		return errors.New(fmt.Sprintf("Error during setting '%s' as active profile: %s", profileName, err.Error()))
```
I believe we might need to change messages for `SetOcContext` to not mixed up with `SetActiveProfile`. 
Probably error messages specific to setting oc context.

Reply:
@budhrg This is intentional. The initial part of the message is same but the err.Error() will be different and hopefully user can understand what happened why and how.

Reply:
Consider this scenario:

I first get message `Profile 'budhrg' set as active profile` which then I assume that my profile is set and after like few seconds I get another message `Error during setting 'budhrg' as active profile: blah blah error message`. I would be confused how I got the first message.